packages/contracts-bedrock/test/L2/L2StandardBridge.t.sol

```diff
@@ -470,6 +500,29 @@ contract L2StandardBridge_Bridge_Test is Bridge_Initializer {
         l2StandardBridge.finalizeDeposit(address(L1Token), address(L2Token), alice, alice, 100, hex"");
     }
 
+    /// @dev Tests that `finalizeDeposit` succeeds when depositing ERC20 with custom gas token.
+    function test_finalizeDeposit_depositingERC20_customGasToken_succeeds() external {
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+
+        vm.mockCall(
+            address(l2StandardBridge.messenger()),
+            abi.encodeWithSelector(CrossDomainMessenger.xDomainMessageSender.selector),
+            abi.encode(address(l2StandardBridge.OTHER_BRIDGE()))
+        );
+
+        vm.expectCall(address(L2Token), abi.encodeWithSelector(OptimismMintableERC20.mint.selector, alice, 100));
+
+        // Should emit both the bedrock and legacy events
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
```

Comment:
technically you can omit all of the bools and only pass address and you get the same effect as all bools being true, reduces amount of code